### PR TITLE
Add Xamarin placeholders for System.Memory package

### PR DESCRIPTION
**Pull request by filipnavara**
Ref: https://github.com/dotnet/runtime/issues/49211

Not really sure how to proceed with servicing OOB package from an ancient branch like this (cc @ViktorHofer).

### CODE REVIEW

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 1]
```diff
@@ -2510,16 +2510,23 @@
...
         "4.5.3",
-        "4.5.4"
+        "4.5.4",
+        "4.5.5"
       ],
-      "BaselineVersion": "4.5.4",
+      "BaselineVersion": "4.5.5",
       "InboxOn": {
         "netcoreapp2.1": "4.1.0.0"
+        "monoandroid10": "Any",
```
Comment (ViktorHofer) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json:2519]: Missing comma above

**Comment by ericstj** [comment]
This is going to [break](https://github.com/dotnet/runtime/issues/26244) older versions of Xamarin which didn't contain System.Memory.  I thought we had agreed that Xamarin was going to fix this: https://github.com/xamarin/xamarin-macios/issues/5687#issuecomment-573223744

There are plenty of other packages that rely on the behavior of the framework winning over NuGet packages...  If that's not happening for System.Memory it needs to be fixed in the SDK.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 1 above)
Comment (filipnavara) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json:2519]: Thanks, fixed.

**Comment by filipnavara** [comment]
@ericstj I am fine with ditching this PR if a preferred fix is to bump the Mono library versions. The issue for this specific package came up several times in the last month and there was no traction in getting it resolved so I tried to naïvely come up with a PR. We use a workaround for this for more than a year now and I'd prefer to solve it properly.

**Comment by ericstj** [comment]
I believe the proper fix is to make the SDK / Xamarin runtime prefer its copy, based on conflict resolution rules.  Since it's never acceptable to prefer the package version here, since it won't unify, that means that the framework version should have a higher assembly version than the package and always be preferred.  My understanding from the [linked issue](https://github.com/xamarin/xamarin-macios/issues/5687#issuecomment-573773826) and @marek-safar's comment was that this was already the case, so further investigation to that is required.  If I had to guess, it's because the way that Xamarin's SDK is surfacing the facades to MSBuild is making them not seen by conflict resolution (just a guess), however I would still expect the runtime to prefer it's own, higher version.

**Pull request closed without merging**